xgboost_ray/data_sources/data_source.py

```diff
@@ -0,0 +1,99 @@
+from enum import Enum
+from typing import Any, Optional, Sequence, Tuple
+
+import pandas as pd
+
+
+class RayFileType(Enum):
+    """Enum for different file types (used for overrides)."""
+    CSV = 1
+    PARQUET = 2
+    PETASTORM = 3
+
+
+class DataSource:
+    supports_central_loading = True
+    supports_distributed_loading = False
+
+    @staticmethod
+    def is_data_type(data: Any,
+                     filetype: Optional[RayFileType] = None) -> bool:
+        """Check if the supplied data matches this data source.
+
+        Args:
+            data (Any): Data set
+            filetype (Optional[RayFileType]): Optional RayFileType. Some
+                data sources might require that this is explicitly set
+                (e.g. if multiple sources can read CSV files).
+
+        Returns:
+            Boolean indicating if this data source belongs to/is compatible
+                with the data.
+        """
+        return False
+
+    @staticmethod
+    def get_filetype(data: Any) -> Optional[RayFileType]:
+        """Method to help guessing the filetype.
+
+        Should return None if the supplied data type (usually a filename)
+        is not covered by this data source, otherwise the filetype should
+        be returned.
```

Comment:
```suggestion
        Returns None if the supplied data type (usually a filename)
        is not covered by this data source, otherwise the filetype
        is returned.
```

Reply:
I would stick with a more imperative wording here even though this is an abstract class